mcs/resource_mcs_db_instance.go

```diff
@@ -434,9 +440,29 @@ func resourceDatabaseInstanceRead(d *schema.ResourceData, meta interface{}) erro
 	log.Printf("[DEBUG] Retrieved mcs_db_instance %s: %#v", d.Id(), instance)
 
 	d.Set("name", instance.Name)
-	d.Set("flavor_id", instance.Flavor)
-	d.Set("datastore", instance.DataStore)
+	d.Set("flavor_id", instance.Flavor.ID)
+	d.Set("datastore", flattenDatabaseInstanceDatastore(*instance.DataStore))
+	d.Set("disk_autoexpand", flattenDatabaseInstanceAutoExpand(instance.AutoExpand, instance.MaxDiskSize))
 	d.Set("region", getRegion(d, config))
+	d.Set("size", instance.Volume.Size)
+	d.Set("configuration_id", instance.ConfigurationID)
+
+	BlockStorageV3Client, err := config.BlockStorageV3Client(getRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("error creating OpenStack block storage client: %s", err)
+	}
+	vol, err := volumeGet(BlockStorageV3Client.(*gophercloud.ServiceClient), instance.Volume.VolumeID).Extract()
+	if err != nil {
+		return fmt.Errorf("error retrieving volume type")
+	}
+	d.Set("volume_type", vol.VolumeType)
```

Comment:
Could we get rid of this? It is a parameter for create only, there is no way to change it during instance lifecycle.

Reply:
This parameter is "required", thus terraform needs to have it specified in user's .tf file